### PR TITLE
Unicornify Display results using structured answers and templates

### PR DESCRIPTION
**Pull request by Sloff**
**Intro**
Alters the existing Unicornify module by using structured answers and templates. Fixes an element on issue #1163 

**What does your Instant Answer do?**
It generates a unique unicorn for an email address

**What problem does your Instant Answer solve (Why is it better than organic links)?**
Provides a unique unicorn pic assosiated with an email

**What is the data source for your Instant Answer? (Provide a link if possible)**
http://unicornify.appspot.com

**What are some example queries that trigger this Instant Answer?**
unicornify test@example.com
test@example.com unicornify

**Which existing Instant Answers will this one supercede/overlap with?**
The original unicornify module

**Are you having any problems? Do you need our help with anything?**
Yes, while testing some images are displayed while others are not, but all the links are working. Please see an example pic below under *Bug*

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
*Before*
before

*After*
after

*Bug*
bug

## Checklist
Please place an 'X' where appropriate.

```
- [x] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
- [x] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
- [x] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
- [x] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
- [] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        - [] Google Chrome
        - [x] Firefox
        - [] Opera
        - [] IE 10

    - Windows 7
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 8
        - [] IE 9
        - [] IE 10

    - Windows XP
        - [] IE 7
        - [] IE 8

    - Mac OSX
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] Safari

    - iOS (iPhone)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - iOS (iPad)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - Android
        - [] Firefox
        - [] Native Browser
        - [] Google Chrome
        - [] Opera

```

### CODE REVIEW

**Comment by moollaza** [comment]
>Yes, while testing some images are displayed while others are not, but all the links are working. Please see an example pic below under Bug

@Sloff I can't get any images to break right now. It seems like the image requests are taking a little long to come respond and in that time it looks a little "broken" but AFAI can tell it's working fine :+1:

File: lib/DDG/Goodie/Unicornify.pm [hunk 1]
```diff
@@ -7,26 +7,52 @@ use CGI qw/img/;
...
+            name => "Unicornify",
+            data => {
+                subtitle => "Unique unicorn",
+                title => "$_",
+                url => unicornify_url(email => $_, size => "200"),
+                image => unicornify_url(email => $_, size => "100")
+            },
+            meta => {
+                sourceName => "Unicornify",
+                sourceUrl => $link
```
Comment (moollaza) [inline lib/DDG/Goodie/Unicornify.pm]: Seems like we're not using `$link` anywhere else so we should ditch it and pass the string into `sourceUrl => "http://unicornify.appspot.com/"`

File: lib/DDG/Goodie/Unicornify.pm [hunk 2]
```diff
@@ -7,26 +7,52 @@ use CGI qw/img/;
...
-		.'<br /><a href="' . unicornify_url(email => $_, size => 128) .'">'
-		.'<img src="/iu/?u='.unicornify_url(email => $_, size => "100").'" class="zci--unicornify-img" /></a>'
-		. 'Learn more at <a href="'.$link.'">unicornify.appspot.com</a>';
-
-		return $answer, heading => $heading, html => $html;
+		
+        return "This is a unique unicorn for $_",
+        structured_answer => {
+            id => "unicornify",
+            name => "Unicornify",
```
Comment (moollaza) [inline lib/DDG/Goodie/Unicornify.pm]: Let's change this to "Social" -- we're trying to avoid putting brands in the Tabs

**Comment by Sloff** [comment]
@moollaza, changes made as suggested

**Comment by mintsoft** [comment]
@moollaza this LGTM

**Comment by Sloff** [comment]
I'm going to close and reopen this

**Pull request closed without merging**